mono/metadata/class.c

```diff
@@ -3826,6 +3826,20 @@ mono_class_is_assignable_from_checked (MonoClass *klass, MonoClass *oklass, gboo
 			}
 		}
 
+		/* 
+		 * a is b does not imply a[] is b[] in the case where b is an interface and
+		 * a is a generic parameter that implements that interface,.
+		 */
+
+		if (MONO_CLASS_IS_INTERFACE_INTERNAL (eclass)) {
+			MonoType *eoclass_byval_arg = m_class_get_byval_arg (eoclass);
+			if (mono_type_is_generic_argument (eoclass_byval_arg) && 
+				MONO_CLASS_IMPLEMENTS_INTERFACE (eoclass, m_class_get_interface_id (eclass))) {
```

Comment:
why do we need this last check?  wouldn't we return false for all interfaces - whether or not the gparam implements it.

Reply:
Yeah, not sure what my thought process was yesterday, this is too specific. I probably wanted it to fall through for... some reason, but we can just fail it here and the result will be the same.